shared/chat/inbox/row/add-new-row.js

```diff
@@ -10,87 +10,110 @@ if (!isMobile) {
   KeyHandler = require('../../../util/key-handler.desktop').default
 }
 
-const _AddNewRow = ({
-  onNewChat,
-  onJumpToChat,
-  isLoading,
-}: {
+type Props = {
   onNewChat: () => void,
+  onSetFilter: (filter: string) => void,
   onJumpToChat: () => void,
   isLoading: boolean,
-}) => (
-  <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', minHeight: 48, position: 'relative'}}>
-    <ClickableBox
-      style={{
-        ...globalStyles.flexBoxRow,
-        alignItems: 'center',
-        flexGrow: 1,
-        flexShrink: 0,
-        justifyContent: 'center',
-        paddingLeft: globalMargins.medium,
-        paddingRight: globalMargins.medium,
-      }}
-      onClick={onNewChat}
-    >
-      <Icon type="iconfont-new" style={{color: globalColors.blue, marginRight: 9}} />
-      <Text type="BodyBigLink">New chat</Text>
-    </ClickableBox>
-    {isLoading &&
-      <Box style={{bottom: 0, left: 0, position: 'absolute', right: 0}}>
-        <LoadingLine />
-      </Box>}
-  </Box>
-)
+}
+
+type State = {
+  inEditingMode: boolean,
+  filter: string,
+}
+
+class _AddNewRow extends Component<void, Props, State> {
+  state: State
+
+  constructor(props: Props) {
+    super(props)
+    this.state = {
```

Comment:
I don't think we can get rid of editing mode, since when you click in the box, you're in editing mode even though the filter is empty. Got rid of the filter state, though, since I can just plumb it down from TypedState.